### PR TITLE
Add some components to add tabs in the right sidebar

### PR DESCRIPTION
**Pull request by amieiro**
## Problem

Currently, if you want to see the discussion for the translation, you have to open it in another page. This page has 3 tabs, with the discussions, the history and the "other languages" translation.

Solves https://github.com/GlotPress/gp-translation-helpers/issues/147.

This PR depends on this https://github.com/GlotPress/GlotPress/pull/1531, because it adds a required filter.

## Solution

This PR adds 3 tabs, with the discussions, the history and the "other languages" translation.

image

## Testing instructions

You need to be on the `tabs-sidebar` branch on the https://github.com/GlotPress/GlotPress/pull/1531 PR, because this PR adds a required filter.

Then you can start working with the content on the 4 tabs:
* Meta. You can work as you usually work on the current tab to test the functionality.
* Discuss. You can create new comments and/or reply to the existing ones.
* History. You can check the translation history on this tab.
* Other locales. You can check the translation other locales made to this translation.

### CODE REVIEW

**Review by trymebytes — COMMENTED** [review]
I just tested this but the content of these tabs don't load for me
Screenshot 2022-12-06 at 08 48 25

**Comment by akirk** [comment]
I am seeing the same behavior as @trymebytes, is there some code missing from the PR?

**Comment by amieiro** [comment]
@akirk I have checked with @trymebytes, and it was a problem with the "Details" button, because I only catch the double click on the row, not the click on the "Details" button. I have resolved it in this [commit](https://github.com/GlotPress/gp-translation-helpers/pull/148/commits/8364338c9c08c9e53273b0c6aeb0792a398479e4).

**Comment by trymebytes** [comment]
We can look at adding some padding here as shown below;
Screenshot 2022-12-09 at 11 29 04

**Comment by amieiro** [comment]
> Add paddings in the "Start a new discussion" element

Solved [here](https://github.com/GlotPress/gp-translation-helpers/pull/148/commits/0eab3e1c9375c2b57daad6681b05e53e30d71349). 

image

File: includes/class-gp-sidebar.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * @package gp-translation-helpers
+ * @since 0.0.2
+ */
+class GP_Sidebar {
+
+	public static function init() {
+		add_filter( 'gp_right_sidebar', array( static::class, 'add_tabs' ), 10, 2 );
+	}
+
+	public static function add_tabs( $meta_sidebar, $defined_vars ) {
```
[the remark this answers is not in the document]
Comment (amieiro) [reply includes/class-gp-sidebar.php:24]: Added [here](https://github.com/GlotPress/gp-translation-helpers/pull/148/commits/30d21c80b11577e1e8193d119bfafb3124de09b9).

File: js/editor.js [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+	/**
+	 * Load the content in the tabs (header tab and content) for the opened row.
+	 *
+	 * @param {Object} element The element that triggers the action.
+	 */
+	function loadTabsAndDivs( element ) {
+		var originalId = element.closest( 'tr' ).attr( 'id' ).substring( 8 );
+		var requestUrl = $gp_translation_helpers_editor.translation_helper_url + originalId + '?nohc';
+		$.getJSON( requestUrl, function( data ) {
+			$( '[data-tab="sidebar-tab-discussion-' + originalId + '"]' ).html( 'Discuss(' + data[ 'helper-translation-discussion-' + originalId ].count + ')' );
```
[the remark this answers is not in the document]
Comment (amieiro) [reply js/editor.js]: Solved [here](https://github.com/GlotPress/gp-translation-helpers/pull/148/commits/92d4bb5d0e382b20d9ecb73885eec0e2d963f32f). 

I had deciced to use "Discuss" to use less space, but we have enough to update to "Discussion", so I have updated the text.

**Review by akirk — APPROVED** [review]
The CSS issue is fixed with the latest commit.

**Review by trymebytes — APPROVED** [review]
Tested and works as expected, LGTM!

**Comment by akirk** [comment]
Oh, I forgot to mention one typography issue, could you fix this in a follow-up? There should be a space before the open parenthesis. That is `Discussion (6)` and not `Discussion(6)`.

**Comment by amieiro** [comment]
> Oh, I forgot to mention one typography issue, could you fix this in a follow-up? There should be a space before the open parenthesis. That is `Discussion (6)` and not `Discussion(6)`.

This was made intentionally, because we have few space in the tabs.

**Comment by akirk** [comment]
Would a `&nbsp;` not have been an option, or a `white-space: nowrap`?

**Comment by amieiro** [comment]
> Would a `&nbsp;` not have been an option, or a `white-space: nowrap`?

I have just [added](https://github.com/GlotPress/gp-translation-helpers/commit/664695bdfcac39a6c45324a9bfb64a845bbc1e3d) some `&nbsp;` spaces and it works fine.

**Pull request merged**